code/modules/power/supermatter/supermatter.dm

```diff
@@ -450,3 +474,107 @@
 	gasefficency = 0.15
 	explosion_power = 35
 
+/obj/machinery/power/supermatter_shard/proc/supermatter_pull(turf/center, pull_range = 10)
+	playsound(src.loc, 'sound/weapons/marauder.ogg', 100, 1, extrarange = 7)
+	for(var/atom/P in orange(pull_range,center))
+		if(istype(P, /atom/movable))
+			var/atom/movable/pulled_object = P
+			if(ishuman(P))
+				var/mob/living/carbon/human/H = P
+				H.apply_effect(2, WEAKEN, 0)
+			if(pulled_object && !pulled_object.anchored && !ishuman(P))
+				step_towards(pulled_object,center)
+				step_towards(pulled_object,center)
+				step_towards(pulled_object,center)
+				step_towards(pulled_object,center)
+
+	return
+
+/obj/machinery/power/supermatter_shard/proc/supermatter_anomaly_gen(turf/anomalycenter, type = 1, anomalyrange = 5)
+	message_admins("4 OK")
+	var/turf/L = pick(orange(anomalyrange, anomalycenter))
+	message_admins("5 OK")
+	if(L)
+		message_admins("6 OK")
+		if(type = 1)
+			message_admins("7 OK")
+			var/obj/effect/anomaly/flux/A = new(L)
+			message_admins("8 OK")
+			A.explosive = 0
+			message_admins("9 OK")
+			A.lifespan = 300
+			message_admins("10 OK")
+		else if(type = 2)
+			var/obj/effect/anomaly/grav/A = new(L)
+			A.lifespan = 250
+		else if(type = 3)
+			var/obj/effect/anomaly/pyro/A = new(L)
+			A.lifespan = 200
+
+	return
+
+/proc/supermatter_zap(atom/source, range = 3, power)
```

Comment:
I'm trying to write my own proc here because I don't like the way the tesla bolts move when shot from a static source. They break some stuff, then they always hit the same object over and over. I also wanna add some more effects than just stuns.

I guess you could just tear this out if it ends up being horrifyingly bad.